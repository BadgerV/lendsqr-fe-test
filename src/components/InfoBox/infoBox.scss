//importing variables from global styles
@import "../../utils/global.styles.scss";
.infobox-container {
  display: flex;
  flex-direction: column;
  gap: 0.7rem;

  background-color: white;
  width: 100%;
  padding: 1rem 1rem 1.3rem 1rem;
  box-shadow: 0 0 10px 1px hsla(0, 0%, 0%, 0.071);
  border-radius: 4px;

  .icon {
    display: block;
  }

  .text {
    font-weight: 500;
    font-size: $secondary-font-size;
    color: $secondary-color;
    margin-bottom: 0.1rem;
  }

  .number {
    font-weight: 700;
    color: $primary-color;
    font-size: $font-size-big;
  }
  @media screen and (max-width: 865px) {
    .text {
      font-size: $tertiary-font-size;
    }

    .number {
      font-size: 1.4rem;
    }
  }
  @media screen and (max-width: 361px) {
    .text {
      font-size: 0.6rem;
    }

    .number {
      font-size: 1.3rem;
    }
  }
}
