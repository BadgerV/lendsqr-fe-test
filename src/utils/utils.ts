/* eslint-disable @typescript-eslint/no-explicit-any */

// custom styles for select dropdown
export const dropDownCustomStyles = {
  input: (provided: any) => ({
    ...provided,
    fontFamily: "Work Sans, sans-serif",
    fontSize: "0.8rem",
    height: "0.2rem",
  }),
  singleValue: (provided: any) => ({
    ...provided,
    fontFamily: "Work Sans, sans-serif",
    fontSize: "0.8rem",
  }),
  control: (provided: any, state: { isFocused: boolean }) => ({
    ...provided,
    background: "transparent",
    borderRadius: "8px",
    fontSize: "0.7rem",
    "&:hover": { borderColor: state.isFocused ? "none" : "none" },
    width: "12rem",
    border: "1px solid #545f7d26",
  }),
  indicatorSeparator: () => ({ display: "none" }),
  dropdownIndicator: (provided: any, state: { isFocused: boolean }) => ({
    ...provided,
    color: "#213F7D",
    transform: state.isFocused ? "rotate(180deg)" : null,
    transition: "transform 0.2s ease-in-out",
    scale: "0.8",
  }),
  option: (provided: any, state: { isSelected: boolean }) => ({
    ...provided,
    color: state.isSelected ? "#000" : "#000",
    fontFamily: "Work Sans, sans-serif",
    fontSize: "0.8rem",
  }),
  placeholder: (provided: any) => ({
    ...provided,
    color: "#545F7D",
    fontFamily: "Work Sans, sans-serif",
    fontSize: "0.7rem",
    padding : "0.4rem",
    fontWeight: "300",
  }),
};
