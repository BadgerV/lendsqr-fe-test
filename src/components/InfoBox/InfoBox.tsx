// Importing external stylesheet
import "../../styles/InfoBox/infoBox.scss";

// Importing the InfoBoxProps type from the types file
import { InfoBoxProps } from "../../types";
import { formatNumberWithCommas } from "../../utils/utils";

// Functional component representing an info box
const InfoBox: React.FC<InfoBoxProps> = ({ icon, text, number }) => {
  return (
    // Container for the info box
    <div className="infobox-container" data-testid="info-box-container">
      {/* Icon */}
      <img src={icon} alt="info-icon" className="icon" width={35} height={35} />

      {/* Text */}
      <span className="text">{text}</span>

      {/* Number */}
      <span className="number">{formatNumberWithCommas(number)}</span>
    </div>
  );
};

// Exporting the InfoBox component as the default export
export default InfoBox;
