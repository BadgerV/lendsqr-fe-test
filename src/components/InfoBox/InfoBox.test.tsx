// Import necessary libraries
import { render } from "@testing-library/react";
import InfoBox from "./InfoBox";
import { formatNumberWithCommas } from "../../utils/utils";
import { it } from "vitest";

// Describe block for InfoBox component tests
describe("InfoBox Component Tests", () => {
  // Rendering Test
  it("renders InfoBox correctly", () => {
    // Render the component
    const { container } = render(
      <InfoBox icon="path/to/icon.png" text="Example Text" number={12345} />
    );

    // Check if the rendered component matches the snapshot
    expect(container).toMatchSnapshot();
  });

  // Utility Function Test
  it("formats number with commas correctly", () => {
    // Test case 1: Positive number
    expect(formatNumberWithCommas(1234567)).toBe("1,234,567");

    // Test case 4: Zero
    expect(formatNumberWithCommas(0)).toBe("0");
  });
});
