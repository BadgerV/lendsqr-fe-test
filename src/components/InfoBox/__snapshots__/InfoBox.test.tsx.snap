// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InfoBox Component Tests > renders InfoBox correctly 1`] = `
<div>
  <div
    class="infobox-container"
    data-testid="info-box-container"
  >
    <img
      alt="info-icon"
      class="icon"
      height="35"
      src="path/to/icon.png"
      width="35"
    />
    <span
      class="text"
    >
      Example Text
    </span>
    <span
      class="number"
    >
      12,345
    </span>
  </div>
</div>
`;
